src/hooks/useTimeout.ts

```diff
@@ -20,6 +20,8 @@ export default function useTimeout(callback: () => void, delay?: number) {
     if (delay) {
       timeoutRef.current = window.setTimeout(tick, delay)
       return () => window.clearTimeout(timeoutRef.current as number)
+    } else {
+      return () => {} // makes TS happy to have an explicit return
     }
```

Comment:
The "else" statement isn't necessary since the "if" portion has a return the only time it will get to here is if !delay. So the return could just be after the if statement. But the else doesn't hurt

Reply:
Yeah, I originally wrote it like you suggest, but it didn't seem to clearly indicate intent as this does. Could go either way I guess.